federation/README.md

```diff
@@ -0,0 +1,61 @@
+# Cluster Federation
+
+Kubernetes Cluster Federation enables users to federate multiple
+Kubernetes clusters. Please see the [user guide](http://kubernetes.io/docs/admin/federation/)
+and the [admin guide](http://kubernetes.io/docs/user-guide/federation/federated-services/)
+for more details about setting up and using the Cluster Federation.
+
+# Building Kubernetes Cluster Federation
+
+Please see the [Kubernetes Development Guide](https://github.com/kubernetes/kubernetes/blob/master/docs/devel/development.md)
+for initial setup. Once you have the development environment setup
+as explained in that guide, you also need to install [`jq`](https://stedolan.github.io/jq/download/)
+
+Building cluster federation should be as simple as running:
+
+```shell
+make build do=gen
+```
+
+To deploy clusters and install federation components, edit the
+`config.default.json` file to describe your clusters and run
```

Comment:
It can be any of the clusters. It just installs the federation components in the cluster which is the default context in the kubeconfig. But to make it more predictable I set the first cluster as the default context - https://github.com/madhusudancs/kubernetes-anywhere/blob/federation/federation/entrypoint.sh#L52